### PR TITLE
Update helm readme with autoregistration flow

### PR DESCRIPTION
**Pull request by zaharidichev**
This PR updates the Helm readme with instructions on how to use our new
authentication and autoregistration flow

Signed-off-by: Zahari Dichev <zaharidichev@gmail.com>

### CODE REVIEW

File: README.md
Comment (siggy) [inline README.md:20]: `agent-values.yaml`

File: README.md
Comment (siggy) [inline README.md:24]: `programmatically`

File: README.md
Comment (siggy) [inline README.md:22]: `programmatically`

File: README.md
Comment (siggy) [inline README.md:36]: remove colon.

i'm tempted to document an additional `Install with params` install flow like:
```bash
export BUOYANT_CLOUD_CLIENT_ID=<your client id>
export BUOYANT_CLOUD_CLIENT_SECRET=<your client secret>
export AGENT_NAME=<your agent name>
helm install --create-namespace --namespace buoyant-cloud --set metadata.agentName=$AGENT_NAME --set api.clientID=$BUOYANT_CLOUD_CLIENT_ID --set api.clientSecret=$BUOYANT_CLOUD_CLIENT_SECRET linkerd-buoyant linkerd-buoyant/linkerd-buoyant
```

this flow arguably eliminates the need to run `Obtain values programmatically` (though I think that's a good flow to document as well).

maybe we should document our recommended flow first, and then document the others as alternatives, something like:

```markdown
# Agent Install
## Obtain values from the Bcloud UI
## Install
## Alternative programmatic install flows
### Obtain values programmatically
### Install with params
# Uninstall
```

File: README.md
Comment (siggy) [inline README.md:42]: remove colon

**Review by siggy — APPROVED** [review]
looks good, left some comments around the various ways folks can install.

**Comment by zaharidichev** [comment]
@siggy I saw you already reviewed this one. We discussed this PR over standup with @klingerf and reached the conclusion that we will not add the programmatic flow for now (it also requires a small change in Bcloud query param handling). Therefore I have removed this part. Does that sound fine? Also, are we good to merge this PR once approved or we want to wait and do that closer to the actual agent release ?

**Comment by siggy** [comment]
> @siggy I saw you already reviewed this one. We discussed this PR over standup with @klingerf and reached the conclusion that we will not add the programmatic flow for now (it also requires a small change in Bcloud query param handling). Therefore I have removed this part. Does that sound fine? Also, are we good to merge this PR once approved or we want to wait and do that closer to the actual agent release ?

@zaharidichev this all sounds good, probably better to just provide a "golden path" for install. i'd hold off merging to the `gh-pages` branch until just before we cut the `v0.8.0` agent release, because once your merge this PR, it will automatically update https://helm.buoyant.cloud.

**Pull request merged**
